### ISSUE TITLE
Split repository into org and image name

### ISSUE THREAD

**Issue by mavenraven** [concourse/docker-image-resource#17]
It would nice if this:

```
 name: git-resource-image
  type: docker-image
  source:
    repository: concourse/git-resource
    email: docker@example.com
    username: username
    password: password
```

was like this

```
 name: git-resource-image
  type: docker-image
  source:
    org: concourse
    image_name: git-resource
    email: docker@example.com
    username: username
    password: password
```

so I could do this

```
 name: git-resource-image
  type: docker-image
  source:
    org: {{org-name}}
    image_name: git-resource
    email: docker@example.com
    username: username
    password: password
```

and easily run someone else's pipeline without needing their creds.

**Comment by vito** [comment]
Yeah I went back and forth on this. It's kind of nice to denormalize it but it's also kind of nice to use the more familiar Docker syntax.

But I suppose we already botched that by having a separate `tag` field.

Another issue is that the `A` in `A/B` can either be a username or an organization.

Perhaps we could support two forms:

```yaml
name: my-kewl-image
type: docker-image
source:
  organization: foo # or username? but that's already used...
  name: bar
  tag: 1.0
```

...and...:

```yaml
name: my-kewl-image
type: docker-image
source: {image: foo/bar:1.0}
```

**Comment by mavenraven** [comment]
@vito Looking at this again, maybe the best way to solve this would be to make

```
name: git-resource-image
  type: docker-image
  source:
    repository: {{org-name}}/git-resource
    email:  {{email}}
    username: {{username}}
```

be something that Concourse can use (instead of making the value be either a key or a literal). This use case feels like its abusing the spirit of the `{{}}` syntax though. I'd hate to see a pipeline where half of it is interpolated in the name of premature genericity. Maybe things that could be public but would make sense to be overrode by a different consumer should have a different syntax with a default value?

**Comment by chendrix** [comment]
This final use case should be fixed by concourse/concourse#545
